src/test/java/de/tum/in/www1/artemis/service/AssessmentServiceTest.java

```diff
@@ -199,6 +199,28 @@ public void createModelingExerciseSubmissionAndCalculateScore() {
         assertThat(result.getResultString()).isEqualTo("6 of 7 points");
     }
 
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void testRatedAfterSubmitResultAndDueDateEqualsSubmissionDateOfResult() {
+        TextExercise exercise = createTextExerciseWithSGI(course1);
+        Submission submissionWithoutResult = new TextSubmission();
+        submissionWithoutResult.setSubmissionDate(futureTimestamp);
+        submissionWithoutResult = database.addSubmission(exercise, submissionWithoutResult, "student1");
+        database.addSubmission((StudentParticipation) submissionWithoutResult.getParticipation(), submissionWithoutResult);
+
+        List<Feedback> feedbacks = createFeedback(exercise);
+        var result = new Result();
+        result.setSubmission(submissionWithoutResult);
+        result.setFeedbacks(feedbacks);
+        submissionWithoutResult.addResult(result);
+
+        var calculatedScore = resultRepository.calculateTotalPoints(feedbacks);
```

Comment:
Are you sure you don't need to use the variable here? If you don't need it, removing it would make the code cleaner.  :)